packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js

```diff
@@ -30,6 +30,66 @@ function workerLog(first, ...args) {
 
 workerLog(`supervisor started`);
 
+function makeMeterControl() {
+  let meteringDisabled = 0;
+
+  function isMeteringDisabled() {
+    return !!meteringDisabled;
+  }
+
+  function assertIsMetered(msg) {
+    assert(!meteringDisabled, msg);
+  }
+  function assertNotMetered(msg) {
+    assert(!!meteringDisabled, msg);
+  }
+
+  function runWithoutMetering(thunk) {
+    const limit = globalThis.currentMeterLimit();
+    const before = globalThis.resetMeter(0, 0);
+    meteringDisabled += 1;
+    try {
+      return thunk();
+    } finally {
+      globalThis.resetMeter(limit, before);
+      meteringDisabled -= 1;
+    }
+  }
+
+  async function runWithoutMeteringAsync(thunk) {
+    const limit = globalThis.currentMeterLimit();
+    const before = globalThis.resetMeter(0, 0);
+    meteringDisabled += 1;
+    try {
+      return await thunk();
+    } finally {
+      globalThis.resetMeter(limit, before);
+      meteringDisabled -= 1;
+    }
```

Comment:
Nervous whether this pattern actually works -- can you get away with scoping async operations like this, even if you use `await`?

Reply:
I just checked with @erights and @michaelfig , and the answer is "yes, but don't do it that way". The `finally` is guaranteed to run, and in our case this pattern would be fine, but their recommendation was to use this instead:
```js
  async function runWithoutMeteringAsync(thunk) {
    const limit = globalThis.currentMeterLimit();
    const before = globalThis.resetMeter(0, 0);
    meteringDisabled += 1;
    return Promise.resolve()
      .then(() => thunk())
      .finally(() => {
        globalThis.resetMeter(limit, before);
        meteringDisabled -= 1;
      });
  }
```

The difference is that if `thunk()` throws (synchronously), in my version (`return await thunk()`) the `finally` block will get executed in the same turn, but in theirs (using `Promise.prototype.finally`) the finalizer always runs in its own turn.

I'll update it to use their recommendation.

Reply:
> I just checked with @erights and @michaelfig , and the answer is "yes, but don't do it that way". The finally is guaranteed to run, and in our case this pattern would be fine, but their recommendation was to use this instead:

Just for the record, what did I say "yes" to? The fact that the finally may run in the same or another turn, as you correctly explain, for me means that code does not *work*.

Reply:
"Yes" meant that `finally { }` always runs. even if there's a `return` or `await` in the `try { }`, which I think was the "does JS work the way you seem to think it works" question that @FUDCo was asking. I think we were both uncertain about whether the JS try/finally pattern continues to work after the introduction of async/await, but it does: the general semantics are the same, modulo the timing subtlety you pointed out.

In this case, my code was not depending upon having a turn boundary between the thunk running and the counter being decremented. `runWithoutMeteringAsync` is not being granted to adversarial code (this is entirely within liveslots), so there's not an adversary to perform reentrancy. The `thunk` is also friendly code (if it were adversarial, a trivial Promise.then would let it regain agency after the thunk's Promise had returned). I'm all in favor of using a pattern that's easier to review, and I've updated this PR to do so, but in this case it wouldn't have caused a bug to do it the `return await thunk()` way.